### PR TITLE
PD2372-TagSort

### PR DESCRIPTION
**Pull request by ekiernan**
### Description
Opening PR for testing
----

### Checklist 
If you can answer "yes" to the following items, please add a checkmark next to the appropriate checklist item(s) **and** notify our WARP documentation team by tagging either @ekiernan or @kayleemathews in a comment on this PR.

- [ ] Did you add inputs, outputs, or tasks to a workflow?
- [ ] Did you modify, delete or move: file paths, file names, input names, output names, or task names?
- [ ] If you made a changelog update, did you update the pipeline version number?

### CODE REVIEW

**Review by jessicaway — COMMENTED** [review]
Do we expect that we would have issues if we tried to update all the warp tool dockers instead of just the metrics dockers?

**Comment by ekiernan** [comment]
> Do we expect that we would have issues if we tried to update all the warp tool dockers instead of just the metrics dockers?

Yes, any task with Loom would fail. So definitely Smart-seq would have problems and I think Slideseq would have problems in the Loom task.

**Review by nikellepetrillo — APPROVED** [review]

File: pipelines/skylab/multiome/Multiome.changelog.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# 2.3.0
```
Comment (kayleemathews) [inline pipelines/skylab/multiome/Multiome.changelog.md:1]: Would you mind updating the pipeline version in the Multiome Overview doc?

File: website/docs/Pipelines/Optimus_Pipeline/README.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ slug: /Pipelines/Optimus_Pipeline/README
 
 | Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |
 | :----: | :---: | :----: | :--------------: |
-| [optimus_v6.1.2](https://github.com/broadinstitute/warp/releases?q=optimus&expanded=true) | October, 2023 | Elizabeth Kiernan | Please file GitHub issues in warp or contact [the WARP team](mailto:warp-pipelines-help@broadinstitute.org) |
+| [optimus_v6.2.-](https://github.com/broadinstitute/warp/releases?q=optimus&expanded=true) | November, 2023 | Elizabeth Kiernan | Please file GitHub issues in warp or contact [the WARP team](mailto:warp-pipelines-help@broadinstitute.org) |
```
Comment (kayleemathews) [inline website/docs/Pipelines/Optimus_Pipeline/README.md:10]:
```suggestion
| [optimus_v6.2.0](https://github.com/broadinstitute/warp/releases?q=optimus&expanded=true) | November, 2023 | Elizabeth Kiernan | Please file GitHub issues in warp or contact [the WARP team](mailto:warp-pipelines-help@broadinstitute.org) |
```

File: website/docs/Pipelines/Optimus_Pipeline/Loom_schema.md [hunk 3]
```diff
@@ -43,18 +43,18 @@ The global attributes (unstuctured metadata) in the h5ad apply to the whole file
...
-| `reads_mapped_intronic` | STARsolo and [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The number of reads counted as intron; counted when `sF` tag of BAM file is assigned to a `5`. | 
-| `reads_mapped_intronic_as` | STARsolo and [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The number of reads counted as intron in the antisense direction; counted when `sF` tag of BAM file is assigned to a `6`. |
-| `duplicate_reads` | [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The number of duplicate reads. |
+| `reads_mapped_exonic` | STARsolo and [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The number of unique reads counted as exon; counted when BAM file's `sF` tag is assigned to `1` or `3` and the `NH:i` tag is `1`. |
+| `reads_mapped_exonic_as` | STARsolo and [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The number of reads counted as exon in the antisense direction; counted when the BAM's `sF`  is assigned to a `2` or `4` and the `NH:i` tag is `1`. |
```
Comment (kayleemathews) [inline website/docs/Pipelines/Optimus_Pipeline/Loom_schema.md:47]:
```suggestion
| `reads_mapped_exonic_as` | STARsolo and [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The number of reads counted as exon in the antisense direction; counted when the BAM's `sF` is assigned to a `2` or `4` and the `NH:i` tag is `1`. |
```

File: website/docs/Pipelines/Optimus_Pipeline/Loom_schema.md [hunk 4]
```diff
@@ -43,18 +43,18 @@ The global attributes (unstuctured metadata) in the h5ad apply to the whole file
...
+| `reads_mapped_intronic` | STARsolo and [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The number of unique reads counted as intron; counted when the BAM files's `sF` tag is assigned to a `5` and the `NH:i` tag is `1`. | 
+| `reads_mapped_intronic_as` | STARsolo and [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The number of unique reads counted as intron in the antisense direction; counted when the BAM file's `sF` tag is assigned to a `6` and the `NH:i` tas is `1`. |
+| `duplicate_reads` | [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | Not currently calculated for Optimus output. The number of duplicate reads. |
```
Comment (kayleemathews) [inline website/docs/Pipelines/Optimus_Pipeline/Loom_schema.md:50]:
```suggestion
| `duplicate_reads` | [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | Not currently calculated for Optimus output; number of duplicate reads. |
```

File: website/docs/Pipelines/Optimus_Pipeline/Loom_schema.md [hunk 5]
```diff
@@ -43,18 +43,18 @@ The global attributes (unstuctured metadata) in the h5ad apply to the whole file
...
 |`pct_mitochondrial_molecules`| [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort)| The percentage of molecules from mitochondrial genes detected for this cell. |
 |`reads_mapped_uniquely`|[TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort)| The number of reads mapped to a single unambiguous location in the genome. |
 |`reads_mapped_multiple`|[TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort)|The number of reads mapped to multiple genomic positions with equal confidence. |
 |`spliced_reads`|[TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort)| The number of reads that overlap splicing junctions. |
-|`antisense_reads`|[TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort)| The number of reads that are mapped to the antisense strand instead of the transcribed strand. |
+|`antisense_reads`|[TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort)| Not calculated for Optimus outputs. See reads_mapped_exonic_as or reads_mapped_intronic_as for antisense counts. |
```
Comment (kayleemathews) [inline website/docs/Pipelines/Optimus_Pipeline/Loom_schema.md:57]:
```suggestion
|`antisense_reads`|[TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort)| Not calculated for Optimus outputs; see `reads_mapped_exonic_as` or `reads_mapped_intronic_as` for antisense counts. |
```

File: website/docs/Pipelines/Optimus_Pipeline/Loom_schema.md [hunk 6]
```diff
@@ -79,8 +79,8 @@ The global attributes (unstuctured metadata) in the h5ad apply to the whole file
...
 | `emptydrops_Total` | [dropletUtils](https://bioconductor.org/packages/release/bioc/html/DropletUtils.html) | The total read counts for each barcode; single-cell data will read `NA` if task is unable to detect knee point inflection. Column is not included for data run in the `sn_rna` mode.  |
-| `reads_mapped_intergenic` | STARsolo and [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The number of reads counted as intergenic; counted when `sF` tag of BAM file is assigned to a `7`. |
-| `reads_unmapped` | [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The number of reads that are unmapped. |
+| `reads_mapped_intergenic` | STARsolo and [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The number of reads counted as intergenic; counted when the BAM file's `sF` tag is assigned to a `7` and the `NH:i` tag is `1`. |
+| `reads_unmapped` | [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The total number of reads that are unmapped, counted when the `sF` tag is `0`. |
```
Comment (kayleemathews) [inline website/docs/Pipelines/Optimus_Pipeline/Loom_schema.md:83]:
```suggestion
| `reads_unmapped` | [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The total number of reads that are unmapped; counted when the BAM file's`sF` tag is `0`. |
```

File: website/docs/Pipelines/Optimus_Pipeline/Loom_schema.md [hunk 7]
```diff
@@ -91,12 +91,12 @@ The global attributes (unstuctured metadata) in the h5ad apply to the whole file
 | `Gene` | [GENCODE GTF](https://www.gencodegenes.org/) | The unique `gene_name` provided in the GENCODE GTF file; identical to the `gene_names` attribute. |
 |`gene_names` | [GENCODE GTF](https://www.gencodegenes.org/) | The unique `gene_name` provided in the GENCODE GTF file; identical to the `Gene` attribute. |
 |`n_reads`|[TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort)| The number of reads associated with this gene. |
-|`noise_reads`|[TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort)| The number of reads that are categorized by 10x Genomics Cell Ranger as "noise". Refers to long polymers, or reads with high numbers of N (ambiguous) nucleotides. |
+|`noise_reads`|[TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort)| Not currently calculated for Optimus output. The number of reads that are categorized by 10x Genomics Cell Ranger as "noise". Refers to long polymers, or reads with high numbers of N (ambiguous) nucleotides. |
```
Comment (kayleemathews) [inline website/docs/Pipelines/Optimus_Pipeline/Loom_schema.md:94]:
```suggestion
|`noise_reads`|[TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort)| Not currently calculated for Optimus output; number of reads that are categorized by 10x Genomics Cell Ranger as "noise"; refers to long polymers, or reads with high numbers of N (ambiguous) nucleotides. |
```

File: website/docs/Pipelines/Optimus_Pipeline/Loom_schema.md [hunk 8]
```diff
@@ -91,12 +91,12 @@ The global attributes (unstuctured metadata) in the h5ad apply to the whole file
...
-| `reads_mapped_exonic_as` | STARsolo and [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The number of reads counted as exon in the antisense direction; counted when `sF` tag of BAM file is assigned to a `2` or `4`. |
-| `reads_mapped_intronic` | STARsolo and [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The number of reads counted as intron; counted when `sF` tag of BAM file is assigned to a `5`. | 
-| `reads_mapped_intronic_as` | STARsolo and [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The number of reads counted as intron in the antisense direction; counted when `sF` tag of BAM file is assigned to a `6`. |
+| `reads_mapped_exonic` | STARsolo and [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The number of unique reads counted as exon; counted when BAM file's `sF` tag of BAM file is assigned to `1` or `3` and the `NH:i` tag is `1`. |
```
Comment (kayleemathews) [inline website/docs/Pipelines/Optimus_Pipeline/Loom_schema.md:96]:
```suggestion
| `reads_mapped_exonic` | STARsolo and [TagSort](https://github.com/broadinstitute/warp-tools/tree/develop/tools/TagSort) | The number of unique reads counted as exon; counted when BAM file's `sF` tag is assigned to `1` or `3` and the `NH:i` tag is `1`. |
```

**Review by kayleemathews — CHANGES_REQUESTED** [review]
Made some suggestions for doc updates. Thanks Liz!

File: pipelines/skylab/multiome/Multiome.changelog.md (hunk 1 above)
Comment (ekiernan) [reply pipelines/skylab/multiome/Multiome.changelog.md:1]: Just bumped it! Thanks!

**Comment by ekiernan** [comment]
> Made some suggestions for doc updates. Thanks Liz!

Thanks for reviewing the documentation, Kaylee!

File: website/docs/Pipelines/Multiome_Pipeline/README.md [hunk 9]
```diff
@@ -7,7 +7,7 @@ slug: /Pipelines/Multiome_Pipeline/README
 
 | Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |
 | :----: | :---: | :----: | :--------------: |
-| [Multiome v2.2.2](https://github.com/broadinstitute/warp/releases) | October, 2023 | Kaylee Mathews | Please file GitHub issues in warp or contact the [WARP Pipeline Development team](mailto:warp-pipelines-help@broadinstitute.org) |
+| [Multiome v2.3.0](https://github.com/broadinstitute/warp/releases) | October, 2023 | Kaylee Mathews | Please file GitHub issues in warp or contact the [WARP Pipeline Development team](mailto:warp-pipelines-help@broadinstitute.org) |
```
Comment (kayleemathews) [inline website/docs/Pipelines/Multiome_Pipeline/README.md:10]:
```suggestion
| [Multiome v2.3.0](https://github.com/broadinstitute/warp/releases) | November, 2023 | Kaylee Mathews | Please file GitHub issues in warp or contact the [WARP Pipeline Development team](mailto:warp-pipelines-help@broadinstitute.org) |
```

**Review by kayleemathews — APPROVED** [review]
LGTM! Thanks!

**Comment by ekiernan** [comment]
Updated truth for:
1. Multiome: https://gotc-jenkins.dsp-techops.broadinstitute.org/job/warp-workflow-tests/56190/
2. Optimus: https://gotc-jenkins.dsp-techops.broadinstitute.org/job/warp-workflow-tests/56191/
3. Slideseq: https://gotc-jenkins.dsp-techops.broadinstitute.org/job/warp-workflow-tests/56192/

**Review by jessicaway — APPROVED** [review]

**Pull request merged**